tests/arguments.nix

```diff
@@ -0,0 +1,132 @@
+        # not the vms in reboot loops
+        boot.kernel.sysctl."vm.panic_on_oom" = 0;
+
+        environment.systemPackages = [
+          git
+        ];
+        systemd.services.wharfix = {
+          wantedBy = [ "multi-user.target" ];
+          after = [ "network.target" ];
+          path = [ nix ];
+          environment.NIX_PATH = "nixpkgs=${pkgs.path}";
+          serviceConfig = {
+            Slice = "wharfix.slice";
+            MemoryAccounting = true;
+            MemoryMax = "256M";
+            Environment = "RUST_BACKTRACE=1";
+            ExecStart = ''
+              ${wharfix}/bin/wharfix \
+                --repo ${repo} \
+                --address ${addr} \
+                --port ${builtins.toString port} \
+                --blob-cache-dir ${blob-cache-dir} \
+                --target ${target} \
+                --index-file-path ${indexFileName} \
+                --add-nix-gcroots
+              '';
+          };
+        };
+        systemd.services.wharfix-path = {
+          after = [ "network.target" ];
+          path = [ nix ];
+          environment.NIX_PATH = "nixpkgs=${pkgs.path}";
+          serviceConfig = {
+            Slice = "wharfix.slice";
+            MemoryAccounting = true;
+            MemoryMax = "256M";
+            Environment = "RUST_BACKTRACE=1";
+            ExecStart = ''
+              ${wharfix}/bin/wharfix \
+                --path ${repo} \
+                --address ${addr} \
+                --port ${builtins.toString port} \
+                --blob-cache-dir ${blob-cache-dir} \
+                --target ${target} \
+                --index-file-path ${indexFileName} \
+                --add-nix-gcroots
+              '';
+          };
+        };
+
+        networking.firewall.allowedTCPPorts = [ port ];
+      };
+
+      client = { ... }: {
+        virtualisation.docker.enable = true;
+        virtualisation.docker.extraOptions = "--insecure-registry registry:${builtins.toString port}";
+      };
+
+    };
+
+    testScript = ''
+      start_all()
+
+      registry.wait_for_open_port(${builtins.toString port})
+      registry.wait_for_unit("wharfix.service")
+      client.wait_for_unit("docker.service")
+
+      client.succeed("docker pull registry:${builtins.toString port}/${image}:master")
+
+      # Test --blob-cache-dir
+      registry.succeed("stat ${blob-cache-dir}")
+
+      # Test --target
+      registry.succeed("stat ${target}")
+
+      # Test --add-nix-gcroots
+      registry.succeed("unlink ${blob-cache-dir}/ma/manifest.json")
+      registry.fail("unlink ${blob-cache-dir}/ma/manifest.json")
+
+      registry.succeed("systemctl stop wharfix")
+
+      registry.fail("wharfix")
+      registry.fail("wharfix --index-file-path really-does-not-exist-${indexFileName} -- port ${builtins.toString port}")
+      registry.fail("wharfix --index-file-path ${indexFileName} --address 256.256.256.256 -- port ${builtins.toString port}")
```

Comment:
Actually I just noticed these whitespaces in `-- port` in this line and the one above, sorry :see_no_evil: Are they intentional? Seems you're trying to test something else here. (nonexistent index-file-path, bad address)